cmd/pmgr/main.go

```diff
@@ -1,11 +1,178 @@
 package main
 
 import (
+	"bytes"
+	"flag"
 	"fmt"
-
-	"github.com/hftamer/go-training/pkg/pmgr"
+	"golang.org/x/crypto/bcrypt"
+	"log"
+	"os"
+	"strings"
 )
 
+type userMap map[string]string
+
 func main() {
-	fmt.Println(pmgr.GetQuote())
+	userData := userMap{}
+	filename := "userData.txt"
+	populateUserMapWithDataFromFile(filename, userData)
+	runCommandLineProgram(userData, filename)
+}
+
+func runCommandLineProgram(userData userMap, fileName string) {
+	addHelperFlagText()
+
+	switch os.Args[1] {
+	case "add":
+		validateCommandLineArguments(4)
```

Comment:
I think the expected lengths are wrong here. When I enter `go run main.go add <account> <password>`, I give the program 3 arguments, not 4.

Reply:
correct me if I'm wrong but I thought the first argument was the file path [`go run main.go`] [`add`] [`<account>`] [`<password>`]

Reply:
Yep, but from the point of view of the user, they are only passing in three arguments.